api/src/main/java/io/grpc/RequestValidatorResolver.java

```diff
@@ -0,0 +1,38 @@
+package io.grpc;
+
+import com.google.protobuf.MessageLiteOrBuilder;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class RequestValidatorResolver {
+
+    private List<RequestValidator<MessageLiteOrBuilder>> validators;
+
+    private Map<String, RequestValidator<MessageLiteOrBuilder>> validatorMap;
+
+    public RequestValidatorResolver(List<RequestValidator<MessageLiteOrBuilder>> validators) {
+        this.validators = validators;
+
+        validatorMap = validators.stream()
+                .collect(Collectors.toMap(this::getClassName, it -> it));
+    }
+
+    private String getClassName(RequestValidator<MessageLiteOrBuilder> it) {
+        Type[] genericInterfaces = it.getClass().getGenericInterfaces();
```

Comment:
What's going on here? Why did into the generics; what class are you wanting? I'd have expected to just use `it.getClass()` directly.

Reply:
I want HelloRequest::class when exists this class 
Is there a better way ?

```kotlin
class HelloRequestValidator extends RequestValidator<HelloRequest>
```